server/routes/activities/administration/handlers/regimeChange.ts

```diff
@@ -0,0 +1,159 @@
+import { Request, Response } from 'express'
+import { plainToInstance, Transform } from 'class-transformer'
+import { ValidateNested } from 'class-validator'
+import ActivitiesService from '../../../../services/activitiesService'
+import { createSessionSlots } from '../../../../utils/helpers/activityTimeSlotMappers'
+import { DaysAndSlotsInRegime } from '../../../../utils/helpers/applicableRegimeTimeUtil'
+import { PrisonRegime } from '../../../../@types/activitiesAPI/types'
+import SimpleTime from '../../../../commonValidationTypes/simpleTime'
+
+export class RegimeTimes {
+  @Transform(({ value }) =>
+    Object.keys(value).reduce((acc, k) => acc.set(k, plainToInstance(SimpleTime, value[k])), new Map()),
+  )
+  @ValidateNested()
+  startTimes: Map<string, SimpleTime>
+
+  @Transform(({ value }) =>
+    Object.keys(value).reduce((acc, k) => acc.set(k, plainToInstance(SimpleTime, value[k])), new Map()),
+  )
+  @ValidateNested()
+  endTimes: Map<string, SimpleTime>
+}
+
+export default class RegimeChangeRoutes {
+  constructor(private readonly activitiesService: ActivitiesService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const regimeTimes = await this.activitiesService.getPrisonRegime(user.activeCaseLoadId, user)
+
+    const regimeSlots = await this.getDaysAndSlots(regimeTimes)
+
+    res.render('pages/activities/administration/regime-times', {
+      regimeTimes,
+      regimeSlots,
+    })
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const { startTimes, endTimes }: RegimeTimes = req.body
+
+    const startTimesObj = Array.from(startTimes.keys()).reduce((acc, key) => {
+      acc[key] = startTimes.get(key)
+      return acc
+    }, {})
+
+    const endTimesObj = Array.from(endTimes.keys()).reduce((acc, key) => {
+      acc[key] = endTimes.get(key)
+      return acc
+    }, {})
+    req.body.startTimes = startTimesObj
```

Comment:
Not sure why you are doing this?

Reply:
True, the `startTimes` and `endTimes` are already stored in the body from the `get`